src/vivarium_census_prl_synth_pop/utilities.py

```diff
@@ -572,36 +572,57 @@ def record_metadata_proportions(final_output_dir: Path) -> None:
             dataset_metadata.groupby(groupby_cols)[aggregate_count_cols].sum().reset_index()
         )
 
+        # Denominator columns
+        denominator_columns_mapper = {
+            "row_noise.row_probability_in_households_under_18": "group_rows.row_probability_in_households_under_18",
+            "row_noise.row_probability_in_college_group_quarters_under_24": "group_rows.row_probability_in_college_group_quarters_under_24",
+        }
+
+        denominator_columns = (
+            list(denominator_columns_mapper.values())
+            + [data_keys.METADATA_COLUMNS.NUMBER_OF_ROWS]
+            if set(denominator_columns_mapper.values()).issubset(dataset_metadata.columns)
+            else [data_keys.METADATA_COLUMNS.NUMBER_OF_ROWS]
+        )
+        numerator_columns = [
+            col
+            for col in dataset_metadata.columns
+            if col not in groupby_cols + denominator_columns
+        ]
+        for column in numerator_columns:
+            denominator_column = denominator_columns_mapper.get(
+                column, data_keys.METADATA_COLUMNS.NUMBER_OF_ROWS
+            )
+            # Calculate noise proportions - prevent divide by zero error caused by very
+            dataset_metadata[f"{column}.proportion"] = (
+                dataset_metadata[column] / dataset_metadata[denominator_column]
+            )
+            # This replaces nan caused by divide by zero with 0.0 in cases where there is really small
+            # population.
+            dataset_metadata.loc[
+                dataset_metadata[f"{column}.proportion"].isna(), f"{column}.proportion"
+            ] = 0.0
```

Comment:
This can be simplified to 

```suggestion
            dataset_metadata[f"{column}.proportion"] = (
                dataset_metadata[column] / dataset_metadata[denominator_column]
            ).fillna(0.0)
```

Reply:
lol I totally forgot about .fillna!